controllers/rollingupgrade_controller.go

```diff
@@ -1066,24 +1066,53 @@ func (r *RollingUpgradeReconciler) UpdateInstance(ctx *context.Context,
 	ch <- nil
 }
 
-func requiresRefresh(ec2Instance *autoscaling.Instance, definition *launchDefinition) bool {
+func (r *RollingUpgradeReconciler) getNodeCreationTimestamp(ec2Instance *autoscaling.Instance) time.Time {
+	for _, node := range r.NodeList.Items {
+		tokens := strings.Split(node.Spec.ProviderID, "/")
+		instanceID := tokens[len(tokens)-1]
+		if instanceID == aws.StringValue(ec2Instance.InstanceId) {
+			return node.ObjectMeta.CreationTimestamp.Time
+		}
+	}
+	return time.Time{}
+}
+
+func (r *RollingUpgradeReconciler) requiresRefresh(ruObj *upgrademgrv1alpha1.RollingUpgrade, ec2Instance *autoscaling.Instance,
+	definition *launchDefinition) bool {
+
+	if ruObj.Spec.ForceRefresh {
+		nodeCreationTS := r.getNodeCreationTimestamp(ec2Instance)
+		if nodeCreationTS.Before(ruObj.CreationTimestamp.Time) {
```

Comment:
should you check if nodeCreationTS is not zero? what happens if in `getNodeCreationTimestamp` node is not found and it returns `time.Time{}`? it will evaluate as `0001-01-01 00:00:00 +0000 UTC` and that would be considered before.
Maybe you can refactor `getNodeCreationTimestamp` to return (time.Time, bool) and you can do the if condition using that bool, if the node is not found return `false, time.Time{}`:
```
if ts, ok := r.getNodeCreationTimestamp(ec2Instance); ok {
    < .. logic .. >
}
```

Reply:
Done